source/common/router/header_parser.cc

```diff
@@ -285,10 +296,20 @@ void HeaderParser::evaluateHeaders(Http::HeaderMap& headers,
     const std::string value =
         stream_info != nullptr ? entry.formatter_->format(*stream_info) : entry.original_value_;
     if (!value.empty() || entry.add_if_empty_) {
-      if (entry.formatter_->append()) {
+      switch (entry.append_action_) {
+      case HeaderValueOption::APPEND_IF_EXISTS_OR_ADD:
         headers.addReferenceKey(key, value);
-      } else {
+        break;
+      case HeaderValueOption::ADD_IF_ABSENT:
+        if (auto header_entry = headers.get(key); header_entry.empty()) {
+          headers.addReferenceKey(key, value);
+        }
+        break;
+      case HeaderValueOption::OVERWRITE_IF_EXISTS_OR_ADD:
         headers.setReferenceKey(key, value);
+        break;
+      default:
+        PANIC_DUE_TO_CORRUPT_ENUM;
```

Comment:
Looking back at https://github.com/envoyproxy/envoy/pull/19204, I think we want `PANIC_ON_PROTO_ENUM_SENTINEL_VALUES`.

Reply:
get it.